### PR TITLE
Support Rule context

### PR DESCRIPTION
**Pull request by aounhaider1**
fixes https://github.com/eclipse/smarthome/issues/2764
Signed-off-by: Aoun Bukhari <bukhari@itemis.de>

### CODE REVIEW

**Comment by kaikreuzer** [comment]
Could you please also adapt https://github.com/eclipse/smarthome/blob/master/docs/documentation/development/bindings/xml-reference.md#supported-contexts?

And add the filter to the "Channel Trigger" definition and the rule context to the Rule Enablement Action, so that both use cases described in the issue are directly working?

**Comment by aounhaider1** [comment]
Updated. I added the the new contexts to "Supported Contexts" definition.

File: docs/documentation/development/bindings/xml-reference.md [hunk 1]
```diff
@@ -127,6 +129,14 @@ In the case of above filter only those items will be shown that satisfy the filt
 
 ```
 (type=Switch OR type=Dimmer) AND (tag=Light OR tag=Heating) 
+
+```
+Similarly, the <strong>Channel</strong> context can contain criteria to filter channels based on <strong>kind</strong> field. The value of <strong>kind</strong> can either be STATE or TRIGGER. For Example:
```
Comment (maggu2810) [inline docs/documentation/development/bindings/xml-reference.md]: For example (lower case "e")

**Comment by kaikreuzer** [comment]
Sorry, both modules are still not working for me. :-( Tested with Chrome and Safari.

You also missed to address this comment:

> And add the filter to the "Channel Trigger" definition and the rule context to the Rule Enablement Action, so that both use cases described in the issue are directly working?

I would think that if you do this, you will have the same results as I have.

File: bundles/automation/org.eclipse.smarthome.automation.module.core/ESH-INF/automation/moduletypes/RuleEnableModuleType.json [hunk 2]
```diff
@@ -28,7 +28,8 @@
 					"label": "Rules UIDs.",
 					"description": "Specifies the rules that should be enabled or disabled.",
 					"required": true,
-					"multiple": true
+                    "multiple": true,
```
Comment (kaikreuzer) [inline bundles/automation/org.eclipse.smarthome.automation.module.core/ESH-INF/automation/moduletypes/RuleEnableModuleType.json]: could you replace spaces by tabs?

**Pull request merged**